datalad/core/local/run.py

```diff
@@ -394,17 +395,28 @@ def _unlock_or_remove(dset_path, paths):
             lgr.debug("Filtered out non-existing path: %s", path)
 
     if existing:
-        remove = Remove()
+        notpresent_results = []
+        # Note: If Unlock() is given a directory (including a subdataset) as a
+        # path, files without content present won't be reported, so those cases
+        # aren't being covered by the "remove if not present" logic below.
         for res in Unlock()(dataset=dset_path, path=existing,
                             on_failure="ignore"):
-            if res["status"] == "impossible":
-                if "cannot unlock" in res["message"]:
-                    for rem_res in remove(dataset=dset_path,
-                                          path=res["path"],
-                                          check=False, save=False):
```

Comment:
note (to myself) -- `check=False` implies that no files are dropped (see e.g. https://github.com/datalad/datalad/issues/1823) and thus replacement with plain `rm` should not introduce change in behavior of absent dropping